yb-voyager/src/srcdb/oracle.go

```diff
@@ -174,6 +174,60 @@ func (ora *Oracle) ExportSchema(exportDir string) {
 	ora2pgExtractSchema(ora.source, exportDir)
 }
 
+// return list of jsons having index info like index name, index type, table name, column name
+func (ora *Oracle) GetIndexesInfo() *[]utils.IndexInfo {
+	// TODO(future): once we implement table-list/object-type for export schema
+	// we will have to filter out indexes based on tables or object types that are not being exported
+	query := fmt.Sprintf(`SELECT INDEX_NAME, INDEX_TYPE, TABLE_NAME FROM ALL_INDEXES
+	WHERE OWNER = '%s'`, ora.source.Schema)
+	rows, err := ora.db.Query(query)
+	if err != nil {
+		utils.ErrExit("error in querying source database for indexes info: %v", err)
+	}
+	defer rows.Close()
+	var indexesInfo []utils.IndexInfo
+	for rows.Next() {
+		var indexName, indexType, tableName string
+		err = rows.Scan(&indexName, &indexType, &tableName)
+		if err != nil {
+			utils.ErrExit("error in scanning query rows for reverse indexes: %v", err)
+		}
+		// skip system indexes or domain indexes (all internal	)
+		if strings.HasPrefix(indexName, "SYS_") || strings.HasPrefix(indexName, "DR$") {
+			continue
+		}
+
+		query = fmt.Sprintf(`SELECT COLUMN_NAME FROM ALL_IND_COLUMNS 
```

Comment:
Would it be simpler to make a single query to fetch indexes as well as its columns?

Reply:
good idea, will do that change.